pyccl/halos/halo_model.py

```diff
@@ -402,31 +402,31 @@ def I_0_22(self, cosmo, k, a,
                 second halo profile. If `None`, `prof1` will be used as
                 `prof2`.
             prof3 (:class:`~pyccl.halos.profiles.HaloProfile`): a
-                second halo profile. If `None`, `prof1` will be used as
+                third halo profile. If `None`, `prof1` will be used as
                 `prof3`.
             prof34_2pt (:class:`~pyccl.halos.profiles_2pt.Profile2pt`):
                 a profile covariance object returning the the
                 two-point moment of `prof3` and `prof4`.
             prof4 (:class:`~pyccl.halos.profiles.HaloProfile`): a
-                second halo profile. If `None`, `prof2` will be used as
+                fourth halo profile. If `None`, `prof2` will be used as
                 `prof4`.
 
         Returns:
              float or array_like: integral values evaluated at each
              value of `k`.
         """
-        if (prof3, prof4) == (None, None):
-            prof3, prof4 = prof1, prof2
-        elif (prof3, prof4).count(None) == 1:
-            raise ValueError("prof3 and prof4 should be both defined or None")
+        if prof3 is None:
+            prof3 = prof1
+        if prof4 is None:
+            prof4 = prof2
```

Comment:
I suggest leave the error as it was. In what scenario would someone define (prof1, prof2, prof3) but no prof4? Sounds to me like bad use of the function. We should accept either of these:
- (`prof1`,) --> (`prof1`, `prof1`, `prof1`, `prof1`)
- (`prof1`, `prof2`) --> (`prof1`, `prof2`, `prof1`, `prof2`)
- (`prof1`, `prof2`, `prof3`, `prof4`)

Reply:
This is not what the docstring says, and I do not want to change the API. Besides that, I don't want to be prescriptive about what is good or bad use of a function.